### PR TITLE
Graph Editor: add log when opening a new file

### PR DESCRIPTION
**Pull request by marwie**
Add small log for when a new material is opened making it easier to see where validation errors in the console belong to (when opening multiple files during a session)

See https://github.com/AcademySoftwareFoundation/MaterialX/issues/1398#issuecomment-1683784647

### CODE REVIEW

File: source/MaterialXGraphEditor/Graph.cpp [hunk 1]
```diff
@@ -175,6 +175,7 @@ mx::DocumentPtr Graph::loadDocument(mx::FilePath filename)
     {
         if (!filename.isEmpty())
         {
+            std::cout << ">>> Reading file " << filename.asString() << std::endl;
```
Comment (kwokcb) [inline source/MaterialXGraphEditor/Graph.cpp:178]: Thanks for adding this. I agree with your issue that logging should be in the UI, but better than nothing I guess.

Comment (marwie) [reply source/MaterialXGraphEditor/Graph.cpp:178]: Ideally validation could be visualized in the UI yes but at least with a little log like this we know for sure which file did cause the errors :)

**Review by jstone-lucasfilm — COMMENTED** [review]
@marwie I like the idea behind this proposal, where we have a way of tracking which files are associated with which errors.  My main concern is that we don't currently use the standard output stream for non-error conditions, which gives text in the console window the meaning of important error conditions that the user should look at closely.  By mixing both error and non-error messages in a single stream, my sense is that we would reduce the value of the messages that are currently written out.  To address this, I believe we'll need to improve our logging system, and that's likely a larger change than you're intending to address here.

**Comment by marwie** [comment]
Hello @jstone-lucasfilm thanks for your input. 

Would perhaps a mid-term solution for this be to include e.g. the filename and perhaps a timestamp in error messages (at the end or the beginning of a message)? That way we don't introduce new types of log messages but still maybe improve the current state by providing a bit more context.  

A potential downside to this could be that it negatively affects readability of the actual message and it would require changes in all places that perform logging (I'm not sure how many there are right now)

**Comment by jstone-lucasfilm** [comment]
Let's hold off on this PR for now, as I believe we should build this feature using a logging API, and we'll need some discussion around how this API should be added to the MaterialX libraries.

**Pull request closed without merging**